src/getMigrateDashboards.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  DashboardState as DashboardState50,
+  ContentRecord,
+} from "@activeviam/activeui-sdk-5.0";
+import { DashboardState as DashboardState51 } from "@activeviam/activeui-sdk-5.1";
+import _omit from "lodash/omit";
+import {
+  ErrorReport,
+  MigrateFromVersions,
+  OutcomeCounters,
+} from "./migration.types";
+import { _addErrorToReport } from "./_addErrorToReport";
+import { _getMapOfFolderIds } from "./_getMapOfFolderIds";
+import { _serializeError } from "./_serializeError";
+
+type MigrateDashboardCallback<fromVersion extends MigrateFromVersions> = (
+  dashboardState: fromVersion extends "5.0" ? DashboardState50 : any,
+) => fromVersion extends "5.0" ? DashboardState51 : any;
```

Comment:
It's equivalent but I would find it clearer this way:
```
) => toVersion extends "5.1" ? DashboardState51 : any;
```

But anyway even though I understand the intent I don't think we should centralize the types here.
Because:
- I think it's a typical case where it's better to colocate the types necessary for each step in the step itself
- In any case, I expect these types to be lost and to require TypeScript workarounds when composing higher up

Rather, we would type things correctly when implementing `migrateDashboard` in each version.
E.g. in `5.0` to `5.1`:
```
function migrateDashboard(legacyDashboardState: DashboardState50): DashboardState51 {...}
```

Reply:
Okay, so you would see the callback type as a union of all migration callbacks?

Reply:
We could type it this way:
```
type MigrateDashboardCallback<FromDashboardState, ToDashboardState> = (dashboardState: FromDashboardState) => ToDashboardState;
```

It's just that this doesn't add a lot of value since it's implicitly `any` for both generics.

What matters is that each `migrateDashboardFunction` under each folder is typed properly, like:
```
function migrateDashboard(legacyDashboardState: DashboardState50): DashboardState51 {...}
```